js/views/message.js

```diff
@@ -70,10 +77,12 @@ views.Message = Backbone.Marionette.ItemView.extend({
 				},
 				type:'POST',
 				success: function () {
-					thisModel.get('flags').set('flagged', !starred);
+					// Retrieve the starred/flag status from the backend and display the results.
+					Mail.UI.messageView.loadMessages();
```

Comment:
No, please don't debug. Continue to point out issues without resolution. I'd love to waste more time. :)